CMakeLists.txt

```diff
@@ -552,17 +552,25 @@ if(ZEPHYR_GIT_INDEX)
 endif()
 
 add_custom_command(
-  OUTPUT ${PROJECT_BINARY_DIR}/include/generated/version.h
+  OUTPUT ${PROJECT_BINARY_DIR}/include/generated/zephyr/version.h
   COMMAND ${CMAKE_COMMAND} -DZEPHYR_BASE=${ZEPHYR_BASE}
-    -DOUT_FILE=${PROJECT_BINARY_DIR}/include/generated/version.h
+    -DOUT_FILE=${PROJECT_BINARY_DIR}/include/generated/zephyr/version.h
```

Comment:
we should probably move all generated headers to `build/include/zephyr`, and include `build/include` (also `build/include/zephyr` to not break immediately).

Reply:
it would be guarded by a Kconfig option, enabled by default now, deprecated later maybe (same as we did for the public includes a while back)

Reply:
@henrikbrixandersen I think the `default y` came from here

Reply:
I see